### ISSUE TITLE
Reading scrolls differs from original

### ISSUE THREAD

**Issue by DominusExult** [exult/exult#497]
> I've seen when comparing DOS/Exult reading of scrolls in Serpent Isle (it does not matter if it has Silver Seed or not), that the read scrolls works in different manner.
> I attach some samples to see the issue. As I've seen, the paragraphs of text in the scrolls, in DOS version are written in a "new page" when in Exult version, are written next to the previous one.
> The samples are from the first scroll we found in the hidden cave behind the red tree at the beggining of the game. First, Second and Four pages I think.
> I don't know if this problem persists in books or in Black Gate.

from our old SF tracker https://sourceforge.net/p/exult/bugs/1894/

![Image](https://github.com/user-attachments/assets/c14b810f-85af-491d-971d-bae886218f19)

**Comment by NotStanley4330** [comment]
I can look at this as well. Should just be a matter of making sure paragraphs breaks are held over to the next page

**Comment by DominusExult** [comment]
I wonder why this no one ever really noticed :)
Because this also affects BG and end of page/end of line is wildly different from what the code in gumps/Text_gump.cc:59 uses. AND AFAICT it is different between BG and SI.

In BG:
- end of page is ' ~~' (hex 207E7E)
- line break is ~~ (7E7E) without space before or after

in SI:
- end of page is "  " (2020) or "~~  " (7E7E002020) (and sometimes an empty page is added which I couldn't figure out why)
- line break is "~" (7E) (no space before or after),"~ " (7E20) or "  " (2020)
- double line break is ~~ without space before or after

**Comment by wench** [comment]
I suspect Text Gump was written to handle books correctly and it sort of worked for scrolls too and left as is.

**Comment by DominusExult** [comment]
yeah, I'm struggling with it but succeeding in getting the scrolls right but it severely messes up the books :)

**Comment by DominusExult** [comment]
I've got a lengthy patch for it. Once you put work in it, it became apparent what is going wrong :)
It's not a 100% faithful recreation of the original, but looks way better than it looks currently. Biggest problem is that in SI scrolls page breaks were managed as embedded nulls (0x00).

[books_n_scrolls.patch](https://github.com/user-attachments/files/18732004/books_n_scrolls.patch)

[Pastebin for easy viewing](https://pastebin.com/3pWNZFMJ)

Collection of original and Exult savegames for BG and SI to compare things:
[savegames.zip](https://github.com/user-attachments/files/18731971/savegames.zip)

Examples of how the text is formatted in USECODE
[books_n_scrolls_examples.txt](https://github.com/user-attachments/files/18732002/books_n_scrolls_examples.txt)

**Comment by marzojr** [comment]
I can try looking over this over the weekend .

**Comment by DominusExult** [comment]
thank you, it really needs looking over as it got out of hand once I tried to clobber it into shape.
Tweaking one thing was easily breaking the other thing.
